src/Host/UserProfile/Impl/ServiceLogger.cs

```diff
@@ -28,23 +27,41 @@ internal sealed class ServiceLogger : ILogger, IDisposable {
             return true;
         }
 
+        
         public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter) {
-            var writer = _writer;
-            if (writer == null) {
-                return;
-            }
-
-            string message = formatter(state, exception);
-            lock (writer) {
-                writer.WriteLine("[{0:u}] <{1}> ({2}):", DateTime.Now, _category, logLevel.ToString()[0]);
-                writer.WriteLine(message);
-
-                if (exception != null) {
-                    writer.WriteLine("Exception: " + exception);
+            lock (_logLock) {
```

Comment:
I preserved this lock from the writer. I don't think it is needed. Please confirm.